modules/json/src/main/scala/gem/json/instances/StaticConfigJson.scala

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.json.instances
+
+import gem.config._
+import io.circe.{ Decoder, Encoder }
+import io.circe.generic.semiauto._
+
+@SuppressWarnings(Array("org.wartremover.warts.PublicInference"))
+trait StaticConfigJson {
+  import enumerated._
+  import offset._
+
+  // GMOS Common
+
+  implicit val GmosShuffleOffsetEncoder: Encoder[GmosConfig.GmosShuffleOffset] = Encoder[Int].contramap(_.detectorRows)
+  implicit val GmosShuffleOffsetDecoder: Decoder[GmosConfig.GmosShuffleOffset] = Decoder[Int].map(GmosConfig.GmosShuffleOffset.unsafeFromRowCount)
+
+  implicit val GmosShuffleCyclesEncoder: Encoder[GmosConfig.GmosShuffleCycles] = Encoder[Int].contramap(_.toInt)
+  implicit val GmosShuffleCyclesDecoder: Decoder[GmosConfig.GmosShuffleCycles] = Decoder[Int].map(GmosConfig.GmosShuffleCycles.unsafeFromCycleCount)
+
+  implicit val GmosNodAndShuffleEncoder: Encoder[GmosConfig.GmosNodAndShuffle] = deriveEncoder
+  implicit val GmosNodAndShuffleDecoder: Decoder[GmosConfig.GmosNodAndShuffle] = deriveDecoder
+
+  implicit val GmosCustomRoiEntryEncoder: Encoder[GmosConfig.GmosCustomRoiEntry] = Encoder[(Short, Short, Short, Short)].contramap(e => (e.xMin, e.yMin, e.xRange, e.yRange))
```

Comment:
When can something be waived away with `deriveEncoder` vs when you actually have to do something?  I suppose in this case `GmosCustomRoiEntry` isn't a normal `case class` but rather a `sealed abstract case class` with controlled construction so it cannot be easily derived.

Reply:
You can use `deriveEncoder` for case classes whose fields already have encoders (so base types; and `Option`s, `List`s, etc., thereof) and for sealed traits whose cases meet the same criteria. But it *won't* try to derive encoders for fields. This prevents the compile-time goose chase that leads to exponential compile times but requires that you manufacture more things explicitly.

You are correct that the issue with `GmosCustomRoiEntry` is that it's not a normal case class so Circe doesn't know how to construct one and take it apart again. So we derive the codec via the relationship with `(Short, Short, Short, Short)` (which we know about but Circe doesn't). I think I'm going to add an optic to move this logic out, since we'll need it again for the doobie `Composite` (and boopickle codec if we go down that path).